cartographer/mapping/internal/3d/pose_graph_3d_test.cc

```diff
@@ -51,16 +65,35 @@ class PoseGraph3DTest : public ::testing::Test {
       return POSE_GRAPH)text";
     auto pose_graph_parameters = test::ResolveLuaParameters(kPoseGraphLua);
     pose_graph_options_ = CreatePoseGraphOptions(pose_graph_parameters.get());
+  }
+
+  void BuildPoseGraph() {
+    auto optimization_problem =
+        common::make_unique<pose_graph::OptimizationProblem3D>(
+            pose_graph_options_.optimization_problem_options());
+    optimization_problem_ = optimization_problem.get();
     pose_graph_ = common::make_unique<PoseGraph3DForTesting>(
-        pose_graph_options_, thread_pool_.get());
+        pose_graph_options_, std::move(optimization_problem),
+        thread_pool_.get());
+  }
+
+  void BuildPoseGraphWithFakeOptimization() {
+    auto optimization_problem =
+        common::make_unique<pose_graph::FakeOptimizationProblem3D>();
+    optimization_problem_ = optimization_problem.get();
+    pose_graph_ = common::make_unique<PoseGraph3DForTesting>(
+        pose_graph_options_, std::move(optimization_problem),
+        thread_pool_.get());
   }
 
   proto::PoseGraphOptions pose_graph_options_;
   std::unique_ptr<common::ThreadPool> thread_pool_;
+  pose_graph::OptimizationProblem3D* optimization_problem_;
```

Comment:
You do not use `optimization_problem_` anywhere? Will you need it in future?

Reply:
Removed. I first tried to mock it for this test and this is the left over.